### PR TITLE
Improve README

### PR DESCRIPTION
**Pull request by DeMoorJasper**
This PR improves the README for anyone to quickly get started with this library, mainly focused at Parcel plugin developers.

Might add other sections if anyone wants to use this library outside of parcel

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,84 @@
 # Parcel's source-map library
 
-A purpose build source-maps library for combining and manipulating source-maps.
+A source-map library purpose build for the Parcel.js bundler with a focus on fast combining and manipulating of source-maps.
```
Comment (mischnic) [inline README.md:3]:
```suggestion
A source map library purpose-build for the Parcel bundler with a focus on fast combining and manipulating of source-maps.
```

**Comment by mischnic** [comment]
Should it mention that emscripten needs to be installed for building the wasm backend?

**Comment by DeMoorJasper** [comment]
@mischnic probably will update accordingly

**Comment by mischnic** [comment]
One last thing: can you change the heading levels so that 'Parcel's source-map library' is the only `h1` heading and then usage, inspiration, contributing are `h2` (and their subsections one level lower, respectively)?

**Pull request merged**